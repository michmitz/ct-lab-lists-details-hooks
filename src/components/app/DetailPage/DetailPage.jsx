import React from 'react'
import PropTypes from 'prop-types';

const DetailPage = ({ name, description, image }) => {
  return <div id="character-details">
        <h2>{name}</h2>
        <p>{description}</p>
        <img src={image}/>
      </div>
}

DetailPage.propTypes = {
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
};

export default DetailPage;
