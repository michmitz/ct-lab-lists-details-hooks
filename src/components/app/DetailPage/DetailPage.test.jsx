import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DetailPage from './DetailPage';

describe('DetailPage component', () => {
  afterEach(() => cleanup());

  it('renders DetailPage', () => {
    const { asFragment } = render(<DetailPage
      name='Michelle'
      description='Coolest new X-Files character'
      image='url@url.com'
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});