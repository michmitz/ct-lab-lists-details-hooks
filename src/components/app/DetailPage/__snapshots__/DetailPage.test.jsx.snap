// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage component renders DetailPage 1`] = `
<DocumentFragment>
  <div
    id="character-details"
  >
    <h2>
      Michelle
    </h2>
    <p>
      Coolest new X-Files character
    </p>
    <img
      src="url@url.com"
    />
  </div>
</DocumentFragment>
`;
